
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="shortcut icon" href="../../assets/images/dna-svgrepo-com.svg">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.0.2">
    
    
      
        <title>IGV and visualisation - NGS - Quality control, Alignment, Visualisation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.38780c08.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.3f72e892.min.css">
        
      
    
    
    
      
        
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
      
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#material" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-header-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M.1 494.1c-1.1 9.5 6.3 17.8 15.9 17.8l32.3.1c8.1 0 14.9-5.9 16-13.9.7-4.9 1.8-11.1 3.4-18.1H380c1.6 6.9 2.9 13.2 3.5 18.1 1.1 8 7.9 14 16 13.9l32.3-.1c9.6 0 17.1-8.3 15.9-17.8-4.6-37.9-25.6-129-118.9-207.7-17.6 12.4-37.1 24.2-58.5 35.4 6.2 4.6 11.4 9.4 17 14.2H159.7c21.3-18.1 47-35.6 78.7-51.4C410.5 199.1 442.1 65.8 447.9 17.9 449 8.4 441.6.1 432 .1L399.6 0c-8.1 0-14.9 5.9-16 13.9-.7 4.9-1.8 11.1-3.4 18.1H67.8c-1.6-7-2.7-13.1-3.4-18.1-1.1-8-7.9-14-16-13.9L16.1.1C6.5.1-1 8.4.1 17.9 5.3 60.8 31.4 171.8 160 256 31.5 340.2 5.3 451.2.1 494.1zM224 219.6c-25.1-13.7-46.4-28.4-64.3-43.6h128.5c-17.8 15.2-39.1 30-64.2 43.6zM355.1 96c-5.8 10.4-12.8 21.1-21 32H114c-8.3-10.9-15.3-21.6-21-32h262.1zM92.9 416c5.8-10.4 12.8-21.1 21-32h219.4c8.3 10.9 15.4 21.6 21.2 32H92.9z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            NGS - Quality control, Alignment, Visualisation
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              IGV and visualisation
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active">
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS - Quality control, Alignment, Visualisation" class="md-nav__button md-logo" aria-label="NGS - Quality control, Alignment, Visualisation">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M.1 494.1c-1.1 9.5 6.3 17.8 15.9 17.8l32.3.1c8.1 0 14.9-5.9 16-13.9.7-4.9 1.8-11.1 3.4-18.1H380c1.6 6.9 2.9 13.2 3.5 18.1 1.1 8 7.9 14 16 13.9l32.3-.1c9.6 0 17.1-8.3 15.9-17.8-4.6-37.9-25.6-129-118.9-207.7-17.6 12.4-37.1 24.2-58.5 35.4 6.2 4.6 11.4 9.4 17 14.2H159.7c21.3-18.1 47-35.6 78.7-51.4C410.5 199.1 442.1 65.8 447.9 17.9 449 8.4 441.6.1 432 .1L399.6 0c-8.1 0-14.9 5.9-16 13.9-.7 4.9-1.8 11.1-3.4 18.1H67.8c-1.6-7-2.7-13.1-3.4-18.1-1.1-8-7.9-14-16-13.9L16.1.1C6.5.1-1 8.4.1 17.9 5.3 60.8 31.4 171.8 160 256 31.5 340.2 5.3 451.2.1 494.1zM224 219.6c-25.1-13.7-46.4-28.4-64.3-43.6h128.5c-17.8 15.2-39.1 30-64.2 43.6zM355.1 96c-5.8 10.4-12.8 21.1-21 32H114c-8.3-10.9-15.3-21.6-21-32h262.1zM92.9 416c5.8-10.4 12.8-21.1 21-32h219.4c8.3 10.9 15.4 21.6 21.2 32H92.9z"/></svg>

    </a>
    NGS - Quality control, Alignment, Visualisation
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../precourse/" class="md-nav__link">
      Precourse preparations
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../course_schedule/" class="md-nav__link">
      Course schedule
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Day 1
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Day 1" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        <span class="md-nav__icon md-icon"></span>
        Day 1
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/intro/" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/sequencing_technologies/" class="md-nav__link">
      Sequencing technologies
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/server_login/" class="md-nav__link">
      Server login
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/quality_control/" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../day1/read_alignment/" class="md-nav__link">
      Read alignment
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-nav__toggle md-toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Day 2
      <span class="md-nav__icon md-icon"></span>
    </label>
    <nav class="md-nav" aria-label="Day 2" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        <span class="md-nav__icon md-icon"></span>
        Day 2
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../file_types/" class="md-nav__link">
      File types
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../samtools/" class="md-nav__link">
      Samtools
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        IGV and visualisation
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="./" class="md-nav__link md-nav__link--active">
      IGV and visualisation
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercises" class="md-nav__link">
    Exercises
  </a>
  
    <nav class="md-nav" aria-label="Exercises">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-a-first-glance-the-e-coli-dataset" class="md-nav__link">
    1. A first glance: the E. coli dataset
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-more-functionality-hg18" class="md-nav__link">
    2. More functionality: hg18
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-hcc1143-data-set" class="md-nav__link">
    3. HCC1143 data set
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../group_work/" class="md-nav__link">
      Group work - organisation
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../day3/" class="md-nav__link">
      Day 3
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exercises" class="md-nav__link">
    Exercises
  </a>
  
    <nav class="md-nav" aria-label="Exercises">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-a-first-glance-the-e-coli-dataset" class="md-nav__link">
    1. A first glance: the E. coli dataset
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-more-functionality-hg18" class="md-nav__link">
    2. More functionality: hg18
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-hcc1143-data-set" class="md-nav__link">
    3. HCC1143 data set
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>IGV and visualisation</h1>
                
                <h2 id="material">Material</h2>
<p>The exercises below are partly based on <a href="https://github.com/griffithlab/rnaseq_tutorial/wiki/IGV-Tutorial">this tutorial</a> from the <a href="https://griffithlab.org/">Griffith lab</a>.</p>
<h2 id="exercises">Exercises</h2>
<h3 id="1-a-first-glance-the-e-coli-dataset">1. A first glance: the E. coli dataset</h3>
<p>Index the alignment that was filtered for the region between 2000 and 2500 kb:</p>
<p><div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~/ecoli/mapping_output
samtools index SRR519926.sorted.region.bam
</code></pre></div>
 Download it together with it&rsquo;s index file (<code>SRR519926.sorted.region.bam.bai</code>) and the reference genome (<code>ecoli-strK12-MG1655.fasta</code>) to your desktop.</p>
<ul>
<li>Load the genome (<code>.fasta</code>) into IGV: <strong>Genomes &gt; Load Genome from File&hellip;</strong></li>
<li>Load the alignment file (<code>.bam</code>): <strong>File &gt; Load from File&hellip;</strong></li>
<li>
<p>Zoom in into the region U00096.3:2046000-2048000. You can do this in two ways:</p>
<ol>
<li>With the search box</li>
</ol>
<p><figure>
  <img src="../../assets/images/search_box.png" width="500"/>
</figure></p>
<ol start="2">
<li>Select the region in the location bar</li>
</ol>
<p><figure>
  <img src="../../assets/images/drag_region.png" width="200"/>
</figure></p>
</li>
<li>
<p>View the reads as pairs, by right click on the reads and select <strong>View as pairs</strong></p>
</li>
</ul>
<p><strong>Exercise:</strong> There are lot of reads that are coloured red. Why is that?</p>
<details class="done"><summary>Answer</summary><p>According to <a href="https://software.broadinstitute.org/software/igv/interpreting_insert_size">IGV</a>, reads are coloured red if the insert size is larger than expected. As you remember, this dataset has a very large variation in insert size.</p>
</details>
<p>Modify the popup text behaviour by clicking on the yellow balloon to <strong>Show Details on Click</strong>:</p>
<figure>
  <img src="../../assets/images/yellow_balloon.png" width="300"/>
</figure>

<p><strong>Exercise:</strong> Click on one of the reads. What kind of information is there?</p>
<details class="done"><summary>Answer</summary><p>Most of the information from the SAM file.</p>
</details>
<p>Colour the alignment by pair orientation by right clicking on the reads, and click <strong>Color alignments by &gt; read strand</strong>.</p>
<h3 id="2-more-functionality-hg18">2. More functionality: hg18</h3>
<p>A lot of model-organism genomes are built-in IGV. Select the human genome version hg18 from the drop down menu:</p>
<figure>
  <img src="../../assets/images/select_hg18.png" width="300"/>
</figure>

<div class="admonition note">
<p class="admonition-title">Other reference genomes</p>
<p>If your genome is not in the short list, more reference genomes can be found in the same drop-down menu at <strong>More..</strong></p>
</div>
<p>Load sample data of the human genome from <strong>File &gt; Load from Server&hellip;</strong>. And select  the SNP validation dataset:</p>
<figure>
  <img src="../../assets/images/select_SNP_annotation.png" width="500"/>
</figure>

<p><strong>Exercise:</strong> Check out the tracks that appeared. What kind of files did we just load?</p>
<details class="done"><summary>Answer</summary><p>We have loaded:</p>
<ul>
<li>a coverage track (WGS Alignments Coverage), generated by IGV based on alignment file.</li>
<li>a sequence alignment track (WGS alignments), probably a BAM file</li>
<li>a track with SNPs (SNP Calls), a bed file</li>
</ul>
<p>The track &ldquo;RefSeq genes&rdquo; was already there and came with the genome.</p>
</details>
<p>Navigate to the gene CLSTN1 by typing it in the search box.</p>
<figure>
  <img src="../../assets/images/gene_search.png" width="500"/>
</figure>

<p>Likewise (i.e. by typing it in the search box), navigate to SNP1 that is annotated in the bed file.</p>
<p>Click on the coverage track where the SNP is:</p>
<figure>
  <img src="../../assets/images/per_base.png" width="300"/>
</figure>

<p><strong>Exercise:</strong> What is the sequence coverage for that base? And the percentage C?</p>
<details class="done"><summary>Answer</summary><p>The coverage is 43, and 17 reads (40%) C.</p>
</details>
<h3 id="3-hcc1143-data-set">3. HCC1143 data set</h3>
<p>For this part, we will be using publicly available Illumina sequence data generated for the
HCC1143 cell line. The HCC1143 cell line was generated from a 52 year old caucasian
woman with breast cancer.</p>
<p>Sequence reads were aligned to version GRCh37 of the human reference genome. We will be working with subsets of aligned reads in the region: chromosome 21: 19,000,000 - 20,000,000.</p>
<p>The BAM files containing these reads for the cancer cell line and the matched normal are:</p>
<ul>
<li><a href="https://xfer.genome.wustl.edu/gxfer1/project/gms/testdata/bams/hcc1143/HCC1143.normal.21.19M-20M.bam"><code>HCC1143.normal.21.19M-20M.bam</code></a></li>
<li><a href="https://xfer.genome.wustl.edu/gxfer1/project/gms/testdata/bams/hcc1143/HCC1143.normal.21.19M-20M.bam.bai"><code>HCC1143.normal.21.19M-20M.bam.bai</code></a></li>
</ul>
<p>First we ensure that IGV is using the same reference genome as our bam file. In this case the reads are aligned to GRCh37, also known as hg19.</p>
<p>Select <strong>Human hg19</strong> from the drop-down list in the top left of the IGV window.</p>
<p>Select <strong>File &gt; Load from File&hellip;</strong> from the main menu and select the BAM file <code>HCC1143.normal.21.19M-20M.bam</code> using the file browser.</p>
<figure>
  <img src="../../assets/images/load_sequence_data.png"/>
</figure>

<p>This BAM file only contains data for a 1 Megabase region of chromosome 21. Let&rsquo;s navigate there to see what genes this region covers. To do so, navigate to <code>chr21:19,000,000-20,000,000</code>.</p>
<p>Navigate to region <code>chr21:19,800,320-19,818,162</code></p>
<p>Load repeat tracks by selecting <strong>File &gt; Load from Server&hellip;</strong> from the main menu and then select <strong>Annotations &gt; Variation and Repeats &gt; Repeat Masker</strong></p>
<p>Right click in the alignment track and select <strong>Color alignments by &gt; insert size and pair orientation</strong></p>
<figure>
  <img src="../../assets/images/repeat_region.png"/>
</figure>

<p><strong>Exercise:</strong> Why are some reads coloured white? What can be the cause of that?</p>
<details class="done"><summary>Answer</summary><p>The white coloured reads have a map quality of 0 (click on the read to find the mapping quality). The cause of that is a LINE repeat region called L1PA3.</p>
</details>
<p>Navigate to region <code>chr21:19,324,500-19,331,500</code></p>
<p>Right click in the main alignment track and select:</p>
<ul>
<li><strong>Expanded</strong></li>
<li><strong>View as pairs</strong></li>
<li><strong>Color alignments by &gt; insert size and pair orientation</strong></li>
<li><strong>Sort alignments by &gt; insert size</strong></li>
</ul>
<figure>
  <img src="../../assets/images/homozygous_deletion.png"/>
</figure>

<p><strong>Exercise:</strong> What is the insert size of the red flagged read pairs? Can you estimate the size of the deletion?</p>
<details class="done"><summary>Answer</summary><p>The insert size is about 2.8 kb. This includes the reads. The deletion should be about 2.5 - 2.6 kb.</p>
</details>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid" aria-label="Footer">
        
          <a href="../samtools/" class="md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
            </div>
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Samtools
              </div>
            </div>
          </a>
        
        
          <a href="../group_work/" class="md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-footer-nav__title">
              <div class="md-ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Group work - organisation
              </div>
            </div>
            <div class="md-footer-nav__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/vendor.77e55a48.min.js"></script>
      <script src="../../assets/javascripts/bundle.9554a270.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: "../..",
          features: [],
          search: Object.assign({
            worker: "../../assets/javascripts/worker/search.4ac00218.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>